File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
#include <string>
using namespace std;

int main() {
	queue<int> Q;
	int test;
	int num;
	string what;

	cin >> test;

	for (int i = 0;i<test;i++) {
		cin >> what;
		if (what == "push") {
			cin >> num;
			Q.push(num);
		}
		else if (what == "pop") {
			if (Q.empty()) cout << "-1\n";
			else if (!Q.empty()) {
				cout << Q.front() << '\n';
				Q.pop();
			}
		}
		else if (what == "size") cout << Q.size()<<'\n';
		else if (what == "empty"){
			if (Q.empty()) cout << "1\n";
			else if (!Q.empty()) cout << "0\n";
		}
		else if (what == "front") {
			if (Q.empty()) cout << "-1\n";
			else if (!Q.empty()) cout << Q.front() << '\n';
		}
		else if (what == "back") {
			if (Q.empty()) cout << "-1\n";
			else if (!Q.empty()) cout << Q.back() << '\n';
		}
	}
}

File: 백준/Silver/10845. 큐/큐.py
import sys

# map(int,sys.stdin.readline().split())
# data = list(map(int,sys.stdin.readline().split()))

n = int(sys.stdin.readline())

queue = []

for x in range(n):
    com = list(sys.stdin.readline().split())

    if com[0] == "push":
        queue.append(com[1])
    elif com[0] == "pop":
        if len(queue)==0:
            print(-1)
        else:
            print(queue[0])
            del queue[0]
    elif com[0] == "size":
        print(len(queue))
    elif com[0] == "empty":
        if len(queue)==0:
            print(1)
        else:
            print(0)
    elif com[0] == "front":
        if len(queue)==0:
            print(-1)
        else:
            print(queue[0])
    elif com[0] == "back":
        if len(queue)==0:
            print(-1)
        else:
            print(queue[-1])
